lasso.R

```diff
@@ -0,0 +1,55 @@
+library(glmnet) # load 'glmnet' package
+
+data <- readRDS("./Data/df_index_demographic_vars.RDS") # read index and demographics data provided by Carolina
+
+data <- data[,-1] # remove 'control', an ID variable
+data$sex <- data$sex-1 # make 'sex' a 0/1 binary variable
+data$data_year <- data$data_year-1997 # normalize 'data_year' to start at 1997
+data$weight <- (data$weight - mean(data$weight)) / sd(data$weight) # standardize `weight`
+
+######################### perform LASSO regression on demographic variables #####################
+# Note: LASSO inherently performs variable selection by shrinking some coefficients to zero
+
+# 1. try with 'grad' (see codebook) - CONTINOUS OUTCOME VARIABLE
+x <- as.matrix(data[,-c(5,6,24,25,26,27,28,29,30)]) # Removes class, educ_cat (redundant), dimension vars, index (both non-demographic)
```

Comment:
@lindechen Try to avoid using index values. As we found out, the data isn't finalized and if one little thing changes when @CarolinaVelasco 's PR comes in, all these indices you are pointing to will be shifted. Dplyr has some great piping that lets your code be legible and it won't be subject to index shifts. For example:

x <- data %>% select(variable1, variable2,..., variableN)

Reply:
That is a good point. I will make the change!